### PR TITLE
PopFactory

### PR DESCRIPTION
**Pull request by IhateTrains**

### CODE REVIEW

File: ImperatorToCK3/Source/Imperator/Provinces/Pops.cpp [hunk 1]
```diff
@@ -17,9 +18,10 @@ void Imperator::Pops::registerKeys()
 		const auto popStr = commonItems::stringOfItem(theStream).getString();
 		if (popStr.find('{') != std::string::npos)
 		{
+			Pop::Factory popFactory;
```
Comment (Idhrendur) [inline ImperatorToCK3/Source/Imperator/Provinces/Pops.cpp:21]: Pops already has a factory as a member, no need to a new factory for each pop

File: ImperatorToCK3/Source/Imperator/Provinces/Pops.h [hunk 2]
```diff
@@ -10,26 +12,29 @@ namespace Imperator
...
-		PopsBloc() = default;
-		explicit PopsBloc(std::istream & theStream);
+		explicit PopsBloc(std::istream& theStream);
 
 		[[nodiscard]] const auto& getPopsFromBloc() const { return pops; }
 
 	private:
 		void registerKeys();
-
+
```
Comment (Idhrendur) [inline ImperatorToCK3/Source/Imperator/Provinces/Pops.h:37]: Spurious tabs.

File: ImperatorToCK3Tests/ImperatorWorldTests/Provinces/PopTests.cpp [hunk 3]
```diff
@@ -1,17 +1,20 @@
 #include "gtest/gtest.h"
 #include "../ImperatorToCK3/Source/Imperator/Provinces/Pop.h"
+#include "../ImperatorToCK3/Source/Imperator/Provinces/PopFactory.h"
 #include <sstream>
 
+
+Imperator::Pop::Factory popFactory;
```
Comment (Idhrendur) [inline ImperatorToCK3Tests/ImperatorWorldTests/Provinces/PopTests.cpp:7]: Either use a test fixture to own the popFactory, or do Imperator::Pop::Factory{}.getPop() in the tests to create a factory. Don't create at global scope.

**Review by Idhrendur — APPROVED** [review]

File: ImperatorToCK3/Source/Imperator/Provinces/Pops.cpp (hunk 1 above)
Comment (IhateTrains) [reply ImperatorToCK3/Source/Imperator/Provinces/Pops.cpp:21]: I intended to remove this one, but apparently forgot.

**Review by Zemurin — APPROVED** [review]

File: ImperatorToCK3/Source/Imperator/Provinces/Pops.h (hunk 2 above)
Comment (IhateTrains) [reply ImperatorToCK3/Source/Imperator/Provinces/Pops.h:37]: There is indentation in empty lines across the project, so this is consistent (not sure if I or the IDE added missing tabs here).

File: ImperatorToCK3Tests/ImperatorWorldTests/Provinces/PopTests.cpp (hunk 3 above)
Comment (IhateTrains) [reply ImperatorToCK3Tests/ImperatorWorldTests/Provinces/PopTests.cpp]: Used `Imperator::Pop::Factory{}.getPop()` ✔️

**Pull request merged**